Require delimiting extra options for invoked commands with '--' (#3497)
systemd/mkosi: mkosi/resources/man/mkosi.1.md

`build`
:   Build the image-based on the settings passed on the command line and in the
    configuration files. This command is the default if no verb is specified.

>     Any arguments specified after the verb will be passed directly
>     to the build script, if one is defined. To pass options to the build
>     script, separate them from regular options with `--`.

Reviewer: These two sentences contradict each other. 
```suggestion
    Arguments may be passed to the build scripts, if some are defined. To pass options to the build
    scripts, separate them from regular mkosi options with `--`.
```

`journalctl`
:   Uses **journalctl** to inspect the journal inside the image.
    Any arguments specified after the **journalctl** verb are appended to the
    **journalctl** invocation. To pass extra options, separate

>     them from regular options with `--`.

Reviewer: ```
All arguments specified after the `journalctl` verb and separated by `--` from the regular options are
appended to the **journalctl** invcation
```
This also changes the formatting of the first journalctl, since that is our verb and not the command.

`coredumpctl`
:   Uses **coredumpctl** to look for coredumps inside the image.
    Any arguments specified after the **coredumpctl** verb are appended to the
    **coredumpctl** invocation. To pass extra options, separate

>     them from regular options with `--`.

Reviewer: ```
All arguments specified after the `coredumpctl` verb and separated from the regular options with `--`
are appended to the **coredumpctl** invcation
```

`sysupdate`
:   Invokes **systemd-sysupdate** with the `--transfer-source=` option set
    to the output directory and the `--definitions=` option set to the
    directory configured with `SysupdateDirectory=`. Any arguments
    specified after the `sysupdate` verb are passed directly to
    **systemd-sysupdate** invocation. To pass extra options, separate

>     them from regular options with `--`.

Reviewer: ```
All arguments specified after the `sysupdate` verb and separated from the regular options with `--` 
are passed directly to **systemd-sysupdate**.
```
